__tests__/proxy.test.ts

```diff
@@ -39,72 +39,112 @@ describe('proxy', () => {
-        let proxyUrl = pm.getProxyUrl(url.parse('https://myserver'));
+        let proxyUrl = pm.getProxyUrl(url.parse('https://github.com'));
+        expect(proxyUrl).toBeDefined();
+    })
+
+    it('getProxyUrl does not return proxyUrl if http_proxy set and in no_proxy list', () => {
+        process.env["http_proxy"] = "http://myproxysvr";
+        process.env["no_proxy"] = "otherserver,myserver,anotherserver:8080"
+        let proxyUrl = pm.getProxyUrl(url.parse('http://myserver'));
         expect(proxyUrl).toBeUndefined();
     })
+
+    it('getProxyUrl returns proxyUrl if http_proxy set and not in no_proxy list', () => {
+        process.env["http_proxy"] = "http://myproxysvr";
+        process.env["no_proxy"] = "otherserver,myserver,anotherserver:8080"
+        let proxyUrl = pm.getProxyUrl(url.parse('http://github.com'));
+        expect(proxyUrl).toBeDefined();
+    })
+
+    it('checkBypass returns true if host as no_proxy list', () => {
+        process.env["no_proxy"] = "myserver"
+        let bypass = pm.checkBypass(url.parse('https://myserver'));
+        expect(bypass).toBeTruthy();
+    })
+
+    it('checkBypass returns true if host in no_proxy list', () => {
+        process.env["no_proxy"] = "otherserver,myserver,anotherserver:8080"
+        let bypass = pm.checkBypass(url.parse('https://myserver'));
+        expect(bypass).toBeTruthy();
+    })
     
-    it('does not return proxyUrl if host in no_proxy list with spaces', () => {
-        process.env["https_proxy"] = "https://myproxysvr";
+    it('checkBypass returns true if host in no_proxy list with spaces', () => {
         process.env["no_proxy"] = "otherserver, myserver ,anotherserver:8080"
-        let proxyUrl = pm.getProxyUrl(url.parse('https://myserver'));
-        expect(proxyUrl).toBeUndefined();
+        let bypass = pm.checkBypass(url.parse('https://myserver'));
+        expect(bypass).toBeTruthy();
     })
     
-    it('does not return proxyUrl if host in no_proxy list with ports', () => {
-        process.env["https_proxy"] = "https://myproxysvr";
+    it('checkBypass returns true if host in no_proxy list with port', () => {
         process.env["no_proxy"] = "otherserver, myserver:8080 ,anotherserver"
-        let proxyUrl = pm.getProxyUrl(url.parse('https://myserver:8080'));
-        expect(proxyUrl).toBeUndefined();
+        let bypass = pm.checkBypass(url.parse('https://myserver:8080'));
+        expect(bypass).toBeTruthy();
     })
-    
-    it('returns proxyUrl if https_proxy set and not in no_proxy list', () => {
-        process.env["https_proxy"] = "https://myproxysvr";
+
+    it('checkBypass returns true if host with port in no_proxy list without port', () => {
+        process.env["no_proxy"] = "otherserver, myserver ,anotherserver"
+        let bypass = pm.checkBypass(url.parse('https://myserver:8080'));
+        expect(bypass).toBeTruthy();
+    })
+
+    it('checkBypass returns true if host in no_proxy list with default https port', () => {
```

Comment:
This test is genuinely new and failed before.

Since the no_proxy list is shared between https_proxy and http_proxy, this scenario should work.